rts/Lua/LuaSyncedRead.cpp

```diff
@@ -2960,6 +2905,36 @@ void ApplyPlanarTeamError(lua_State* L, int allegiance, float3& mins, float3& ma
 	}
 }
 
+/**
+ * @brief This populates the Lua table with the unit IDs of units that are in a
+ * particular region. 
+ * 
+ * @param allegiance The allegiance of the units to add to the table.
+ * @param units The units to add to the table.
+ * @param inRegion A lambda checking if the unit position is in the region
+ */
+template<typename InRegion>
+static void GetFilteredUnits(lua_State *L, int allegiance, const std::vector<CUnit *> &units, InRegion inRegion, bool fullRead) {
+	unsigned int count = 0;
+
+	const int readTeam = CLuaHandle::GetHandleReadTeam(L);
+	const int readAllyTeam = CLuaHandle::GetHandleReadAllyTeam(L);
+
```

Comment:
Given that `fullRead` is in practice always `CLuaHandle::GetHandleFullRead(L)` and that `L` gets passed here, perhaps we could do it like this so as not to have to pass `fullRead`?
```suggestion
	const int readAllyTeam = CLuaHandle::GetHandleReadAllyTeam(L);
	const bool fullRead = CLuaHandle::GetHandleFullRead(L);
```

Reply:
Again this was me not being familiar with how expensive re-reading will be. From the is visible discussion is sounds like it won't be expensive so I'll do that thank you

Reply:
I'll do the same for allegiance while I'm at it